### ISSUE TITLE
Add feature that removes XMP metadata from pdf(s)

### ISSUE THREAD

**Issue by ThiloteE** [JabRef/jabref#8277]
**Problem:**
My goal was to write XMP metadata to a pdf, but there was metadata attached to the pdf already (because i already attached another entry to it once, but that entry had wrong data), so now there is data that i do NOT WANT to have attached + the data that i WANT to have attached.

**Describe the solution you'd like**
Add a feature that allows to remove all XMP metadata from one or multiple pdfs.

**Comment by ThiloteE** [comment]
When i push

```
@Test{test,
  author = {test},
  date   = {2021},
  file   = {:Thompson (2020-03-30) middle-class-remorse-re-embracing-liberal-democracy-in-the-philippines-and-thailand.pdf:PDF},
  title  = {test},
}
```

to the pdf file, there is still some leftover XMP metadata as can be seen here:

![image](https://user-images.githubusercontent.com/73715071/143677511-b67d97b1-dfe9-4c5e-a610-9eed7bb49741.png)

**Comment by ThiloteE** [comment]
Related fruitful discussion on https://gist.github.com/hubgit/6078384, but i am not sure in how far tools like these can be implemented with Jabref.

**Comment by koppor** [comment]
Current "feature" of JabRef: Remove specified fields. Maybe this helps for your case somehow? When listing "language", "number", ... in the fields to clear? (Feature highly requested by @adaerr fiew years ago)

![grafik](https://user-images.githubusercontent.com/1366654/144916494-dec75182-96bf-438c-b3fc-b99da1582989.png)

**Comment by ThiloteE** [comment]
Yes! Indeed, Koppor, this is a step in the right direction. I did some tests:

**These conditions need to be fulfilled for Metadata to be deleted:**

1. `Do not write the following fields to XMP Metadata` needs to be ticked
2. The field that is supposed to be deleted needs to be in the list Koppor posted in https://github.com/JabRef/jabref/issues/8277#issuecomment-987172127 .
3. The field that is supposed to be deleted needs to be in the library data of the entry that is supposed to be written to the linked file.

**Example:**

```
@Test{test,
  author   = {test2},
  date     = {2021},
  file     = {:test/test2 (2021).pdf:PDF},
  language = {korean},
  number   = {2},
  title    = {test},
}
```

If i put `Number` into the list while `Do not write the following fields to XMP Metadata` is ticked, it will delete the metadata instead of writing the number 2.

If i do the same, except removing the number field from the entry (`number = {2},`) it will NOT delete the metadata.

So we have found out how to delete metadata with Jabref and the way it currently works, it allows very fine grained usage. This is good. It is not perfect, but it is good. A possible and low hanging fruit for improvement would be to ease the workflow by cutting down on the conditions that need to be fulfilled to delete something, especially condition 3 seems tedious.

**Comment by ThiloteE** [comment]
The next question i asked myself: How then would it be possible to delete the metadata not only for a single entry but for ALL pdf files i have linked to the entries within my library?

My aim is to substitute all the 'bad' and 'false' metadata that is currently attached to my pdfs with the (maybe not perfect, but at least ... ) more correct metadata i got from importing via DOI and manual corrections.

**Prototype (untested) workaround**

1. Create an entry with ALL fields available.
2. Link ALL pdfs that i want to remove metadata from to this one entry. Probably via `main file directory`.

Problems with workaround:
1. Tedious (File would need to be moved, Fields need to be entered in the list, fields need to be entered in the entry)
2. I am not yet sure what kind of search expression would work that will find ALL files within that directory.
3. Knowledge about this method is needed. New users may not be aware.

**Proper solution would be:**

Delete ALL metadata only from the entries that are already linked to the entries within that specific library by pressing a single button.

**Comment by JeffersonJefferson-pixel** [comment]
Hi, I am new to open source and would like to contribute to the project. Is it okay for me to try working on this issue?

**Comment by JeffersonJefferson-pixel** [comment]
Noted. I will open a draft PR once I make some progress.

**Comment by ThiloteE** [comment]
For testing your changes, I can recommend ExifTool. https://www.exiftool.org/

In the [FAQ](https://www.exiftool.org/faq.html#Q30) it is explained how to extract (read) really all available metadata that is attached to an PDF


> 30. "How do I extract absolutely all metadata from a file?"
> 
>     By default, duplicate tags, unknown tags, embedded tags, and System tags that require external utilities are not extracted. The main reason for this is performance; extracting these tags will significantly increase processing time for some files. The following command extracts everything possible with ExifTool:
> 
>     `exiftool -ee3 -U -G3:1 -api requestall=3 -api largefilesupport FILE`
> 
>     (The -G3:1 option is included in the above command only to give an indication of where the metadata was stored.)

**Comment by koppor** [comment]
Some code was done at https://github.com/JabRef/jabref/pull/8681, however, the contributors did not continue working on it. Potential contributors can use the code and the discussions as basis.

**Comment by koppor** [comment]
This is a `CleanUpJob` similar to org.jabref.logic.cleanup.MoveFilesCleanup.

**Comment by ganesh-vk** [comment]
I've looked at the implementation proposed in #8681 and I have a few questions.


> Current "feature" of JabRef: Remove specified fields. Maybe this helps for your case somehow? When listing "language", "number", ... in the fields to clear? (Feature highly requested by [@adaerr](https://github.com/adaerr) few years ago)
> 
> ![grafik](https://user-images.githubusercontent.com/1366654/144916494-dec75182-96bf-438c-b3fc-b99da1582989.png)

Is it necessary to use the same preferences when a user wants to delete the metadata? In the aforementioned PR the implementation did not delete metadata if it was not selected, and in the follow up it was suggested that [all metadata](https://github.com/JabRef/jabref/pull/8681#issuecomment-1105658347) should be deleted. Would the ideal solution delete all metadata if the setting is not enabled, and only delete those checked if it is enabled, or is there another solution that would fit better? I assume implementing a solution removes all metadata is fairly trivial but I could be wrong. 

In [comment](https://github.com/JabRef/jabref/pull/8681#issuecomment-1105853549), it is suggested that another settings like field is created for XMP metadata deletion, this seems counter intuitive to me since both serve a very similar purpose but I might be missing something. 

Do we delete the metadata from all files in the database context or give the user the ability to select files?

I was also wondering if it would be possible to place this functionality in `LinkedFilesEditor` since it seemed the most appropriate, the writing metadata functionality is present there as well. 

I'd like to work on this issue once I understand these decisions. Thank you!

**Comment by koppor** [comment]
> Do we delete the metadata from all files in the database context or give the user the ability to select files?

For the latter, I asked for Cleanup Job

**Comment by koppor** [comment]
> Is it necessary to use the same preferences when a user wants to delete the metadata?

Three options are discussed:

- Delete all XMP meta data (in the PR)
- Delete XMP meta data configured in the preferences
- Delete XMP meta data potentially been written by JabRef

DevCall decision: option A is fine for us.
